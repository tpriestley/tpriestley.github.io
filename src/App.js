import React, { createContext, useReducer } from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { createGlobalStyle, ThemeProvider } from "styled-components";


import Home from './components/Home'
import About from './components/About'
import Error from './components/Error'
import Nav from "./components/Nav"
import API from "./components/API"


import { initialState, Reducer } from "./components/Reducer";

export const AppContext = createContext();

function App() {
  const [state, dispatch] = useReducer(Reducer, initialState);
  const { currentTheme } = state;
  return (
    <div className="todoapp stack-large">
      <ThemeProvider theme={currentTheme}>
      <AppContext.Provider value={{ ...state, dispatch }}>
          <GlobalStyles />
          <Nav />
          <BrowserRouter>
          <Switch>
            <Route path='/' component={Home} exact />
            <Route path='/about' component={About} />
            <Route path='/api' component={API} />
            <Route component={Error} />
          </Switch>
          </BrowserRouter>
          </AppContext.Provider>
        </ThemeProvider>
    </div>
  );
}

const GlobalStyles = createGlobalStyle`
  html, body {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
    background: ${props => props.theme.backgroundColor};
    color: ${props => props.theme.textColor};
    font-family: sans-serif;
  }
`;

export default App;