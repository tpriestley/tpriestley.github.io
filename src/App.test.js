import React from 'react';
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom';
import App from './App';

const DATA = [
    { id: "todo-0", name: "Eat", completed: true },
    { id: "todo-1", name: "Sleep", completed: false },
    { id: "todo-2", name: "Repeat", completed: false }
  ];


describe('Array and Object Tests', () => {
    test('Property and Boolean Test', () => {
        expect(DATA[0]).toHaveProperty('completed', true)
    });
    
    test('Task Array Count', () => {
        expect(DATA.length).toBe(3)
    });
});

describe('String Tests', () => {
    const container = document.createElement("div");
    ReactDOM.render(<App tasks={DATA}/>, container);
    test('Value of Tasks Remaining', () => {
        expect(container.querySelector("h3").textContent).toBe("3 tasks remaining")
    });
    test('Value of Tasks Remaining', () => {
        expect(container.querySelector("h2").textContent).toBe("What needs to be done?")
    });
    test('Value of Add Button', () => {
        expect(container.querySelector("button").textContent).toBe("Add")
    });
});

describe('Snapshot Test', () => {
    test('App snapshot test', () => {
        const component = renderer.create(<App tasks={DATA}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});