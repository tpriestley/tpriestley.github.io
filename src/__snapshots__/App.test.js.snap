// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test App snapshot test 1`] = `
<div
  className="todoapp stack-large"
>
  <form
    onSubmit={[Function]}
  >
    <h2
      className="label-wrapper"
    >
      <label
        className="label__lg"
        htmlFor="new-todo-input"
      >
        What needs to be done?
      </label>
    </h2>
    <input
      autoComplete="off"
      className="input input__lg"
      id="new-todo-input"
      name="text"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="btn btn__primary btn__lg"
      type="submit"
    >
      Add
    </button>
  </form>
  <div
    className="filters btn-group stack-exception"
  >
    <button
      aria-pressed={true}
      className="btn toggle-btn"
      onClick={[Function]}
      type="button"
    >
      <span
        className="visually-hidden"
      >
        Show 
      </span>
      <span>
        All
      </span>
      <span
        className="visually-hidden"
      >
         tasks
      </span>
    </button>
    <button
      aria-pressed={false}
      className="btn toggle-btn"
      onClick={[Function]}
      type="button"
    >
      <span
        className="visually-hidden"
      >
        Show 
      </span>
      <span>
        Active
      </span>
      <span
        className="visually-hidden"
      >
         tasks
      </span>
    </button>
    <button
      aria-pressed={false}
      className="btn toggle-btn"
      onClick={[Function]}
      type="button"
    >
      <span
        className="visually-hidden"
      >
        Show 
      </span>
      <span>
        Completed
      </span>
      <span
        className="visually-hidden"
      >
         tasks
      </span>
    </button>
  </div>
  <h3
    id="list-heading"
  >
    3 tasks remaining
  </h3>
  <ul
    aria-labelledby="list-heading"
    className="todo-list stack-large stack-exception"
    role="list"
  >
    <li
      className="todo"
    >
      <div
        className="stack-small"
      >
        <div
          className="c-cb"
        >
          <input
            defaultChecked={true}
            id="todo-0"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="todo-label"
            htmlFor="todo-0"
          >
            Eat
          </label>
        </div>
        <div
          className="btn-group"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            Edit 
            <span
              className="visually-hidden"
            >
              Eat
            </span>
          </button>
          <button
            className="btn btn__danger"
            onClick={[Function]}
            type="button"
          >
            Delete 
            <span
              className="visually-hidden"
            >
              Eat
            </span>
          </button>
        </div>
      </div>
    </li>
    <li
      className="todo"
    >
      <div
        className="stack-small"
      >
        <div
          className="c-cb"
        >
          <input
            defaultChecked={false}
            id="todo-1"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="todo-label"
            htmlFor="todo-1"
          >
            Sleep
          </label>
        </div>
        <div
          className="btn-group"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            Edit 
            <span
              className="visually-hidden"
            >
              Sleep
            </span>
          </button>
          <button
            className="btn btn__danger"
            onClick={[Function]}
            type="button"
          >
            Delete 
            <span
              className="visually-hidden"
            >
              Sleep
            </span>
          </button>
        </div>
      </div>
    </li>
    <li
      className="todo"
    >
      <div
        className="stack-small"
      >
        <div
          className="c-cb"
        >
          <input
            defaultChecked={false}
            id="todo-2"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="todo-label"
            htmlFor="todo-2"
          >
            Repeat
          </label>
        </div>
        <div
          className="btn-group"
        >
          <button
            className="btn"
            onClick={[Function]}
            type="button"
          >
            Edit 
            <span
              className="visually-hidden"
            >
              Repeat
            </span>
          </button>
          <button
            className="btn btn__danger"
            onClick={[Function]}
            type="button"
          >
            Delete 
            <span
              className="visually-hidden"
            >
              Repeat
            </span>
          </button>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
